northstar/src/runtime/island/seccomp.rs

```diff
@@ -89,7 +89,7 @@ impl AllowList {
 
         let sf_prog = sock_fprog {
             len: self.list.len() as u16,
-            filter: self.list.as_mut_ptr(),
+            filter: self.list.as_ptr() as *mut bindings::sock_filter,
```

Comment:
How can a non-mut pointer be cast into a mut pointer like that?

Reply:
[Here](https://internals.rust-lang.org/t/what-is-the-real-difference-between-const-t-and-mut-t-raw-pointers/6127/3) is the full story about it.

Reply:
If I understand it correctly, calling `apply()` twice would now cause two mutable pointers to `self.list` to exist. It sounds similar to this bug in `as_mut_ref` that was fixed in Rust 1.15.1: https://blog.rust-lang.org/2017/02/09/Rust-1.15.1.html

They went ahead and changed the parameter type to `&mut self`.